src/CMakeLists.txt

```diff
@@ -629,6 +629,17 @@ target_link_libraries(rados_snap_set_diff_obj
 
 option(WITH_LIBRADOSSTRIPER "build with libradosstriper support" ON)
 
+if(CMAKE_VERSION VERSION_LESS "3.20")
+  # see src/common/options/CMakeLists.txt
+  set(all_options
+    global cephfs-mirror crimson mgr mds mds-client mon osd rbd rbd-mirror immutable-object-cache ceph-exporter)
+  foreach(name ${all_options})
+    set_property(SOURCE
+      ${PROJECT_BINARY_DIR}/include/${name}_legacy_options.h
+      PROPERTY GENERATED TRUE)
```

Comment:
This still fails for me with a slew of errors like in https://gitlab.kitware.com/cmake/cmake/-/issues/18399:
```
-- Configuring done
CMake Error at cmake/modules/BuildBoost.cmake:307 (_add_executable):
  Cannot find source file:

    /home/dis/src/ceph/build/include/rgw_legacy_options.h

  Tried extensions .c .C .c++ .cc .cpp .cxx .cu .m .M .mm .h .hh .h++ .hm
  .hpp .hxx .in .txx
Call Stack (most recent call first):
  src/CMakeLists.txt:720 (add_executable)


CMake Error at cmake/modules/BuildBoost.cmake:299 (_add_library):
  Cannot find source file:

    /home/dis/src/ceph/build/include/global_legacy_options.h

  Tried extensions .c .C .c++ .cc .cpp .cxx .cu .m .M .mm .h .hh .h++ .hm
  .hpp .hxx .in .txx
Call Stack (most recent call first):
  src/auth/CMakeLists.txt:23 (add_library)


CMake Error at cmake/modules/BuildBoost.cmake:299 (_add_library):
  Cannot find source file:

    /home/dis/src/ceph/build/include/global_legacy_options.h

  Tried extensions .c .C .c++ .cc .cpp .cxx .cu .m .M .mm .h .hh .h++ .hm
  .hpp .hxx .in .txx
Call Stack (most recent call first):
  src/common/CMakeLists.txt:184 (add_library)

[...]

-- Generating done
CMake Generate step failed.  Build files cannot be regenerated correctly.
```
All of them are coming from `cmake/modules/BuildBoost.cmake`, line 299 or 307:
```
297 # override add_library() to add Boost headers dependency
298 function(add_library target)                 
299   _add_library(${target} ${ARGN})      
300   # can't add dependencies to aliases or imported libraries
301   if (NOT ";${ARGN};" MATCHES ";(ALIAS|IMPORTED);")           
302     maybe_add_boost_dep(${target})     
303   endif()
304 endfunction()                      
305        
306 function(add_executable target)
307   _add_executable(${target} ${ARGN})                    
308   maybe_add_boost_dep(${target})                          
309 endfunction()
```